readers/mocks/messages.go

```diff
@@ -24,29 +24,27 @@ func NewMessageRepository(messages map[string][]readers.Message) readers.Message
 	}
 }
 
-func (repo *messageRepositoryMock) ReadAll(chanID string, offset, limit uint64, query map[string]string) (readers.MessagesPage, error) {
+func (repo *messageRepositoryMock) ReadAll(chanID string, rpm readers.PageMetadata) (readers.MessagesPage, error) {
```

Comment:
This mock does not work properly because it ignores all the parameters other than offset and limit. Other parameters should probably be added into account and used for better API testing (Currently, we don't check returned messages, but only response code in API tests).

Reply:
Looks like the `total` value is missing in the mock? Also, in this implementation, it looks like the operator between filters is `OR`, rather than `AND`.

Reply:
Right, thank you. I just added Total check and modified the mock to allow combination of filters.